docs/DeveloperGuide.md

```diff
@@ -1,14 +1,157 @@
 # Developer Guide
 
-## Design & implementation
+## Design
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+## Implementation
+This section describes some noteworthy details on how certain features are implemented.
+
+### View One Shift feature
+#### Implementation
+The current implementation of the View One Shift feature is to call the `App Controller` from `Duke`, and when the user requests to View One Shift, they input "view one shift" as a command.
+The `App Controller` then calls the method from its class.
+
+Within the `View One Shift` method, the user is prompted to provide details regarding the date and shift index they are looking for.
+The `App Controller` will then call some methods in Shift to verify the shift exists, and returns the employees scheduled if any.
+
+Given below is an example usage scenario and how this method works at each step.
+This assumes that the data has already been loaded (`date: `11/04/2021, `shiftIndex:` 1, `vacancy:` 2, `employees` : [Adam, Eve])
+
+|Step Number| System Output| User Input | Remarks|
+|-------------|--------------|------------------|------------------|
+|1|Enter Command: | view one shift | user can type in `help` for a list of commands|
+|2|Enter Shift date (in dd/MM/YYYY):|11/04/2021|date format is specified|
+|3|Enter Shift index | 1 | Shift index is from 1-6, each index representing a 4 hr block|
+|4|The people assigned to the shift are: [Adam,Eve]|-|Empty array will be returned if no employee is assigned for the shift|
+
+Step 1: The user launches the program, with the data already loaded. The application then prompts the user for their command.
+
+Step 2: The user inputs a date, and the date is processed by a DateTimeFormatter to ensure that it is in the correct format. 
+This date will be added into the shift ArrayList. 
+
+Step 3: The user inputs a shift index, which represents which shift the user is looking for. For instance, index 1 refers to 
+0000-0400 in the morning, which will be relevant to 24hr fast food chains such as McDonalds!
+
+
+Given below is what will be shown if wrong/invalid inputs are given
+
+|Step Number| System Output| Wrong User Input | System Output|
+|-------------|--------------|------------------|------------------|
+|1|Enter Command: | view one Shift | invalid command|
+|2|Enter Shift date (in dd/MM/YYYY):|12/05/2021|Date chosen has no shifts|
+|3|Enter Shift index | 1 | Shift index selected is not available|
+
+Given below is a sequence diagram of the View One Shift feature
+![img.png](img.png)
+
+### add schedule to employee feature
+#### Implementation
+
+The add schedule feature uses of the `Employee` class in the `employee` package to store the schedules of each employee. 
+
+The class diagram for the `Employee` class is as follows:
+
+![employee class diagram v2.PNG](employee%20class%20diagram%20v2.PNG)
```

Comment:
![image](https://user-images.githubusercontent.com/40291462/113528098-29bfec00-95f2-11eb-8b01-fb0dc2f4aeb3.png)

You may have missed out on the return values for some of your functions that seem to return Boolean values (e.g. isScheduleValid, isInteger). Also, do note the misspelling of "String" for setName().